Apps/WebClient/src/NewClientApp/src/stores/note.ts

```diff
@@ -0,0 +1,197 @@
+        lastOperation.value = new Operation(
+            note.id as string,
+            OperationType.UPDATE
+        );
+    }
+
+    function setDeletedNote(note: UserNote) {
+        const noteIndex = notes.value.findIndex((x) => x.id === note.id);
+        if (noteIndex > -1) {
+            lastOperation.value = new Operation(
+                note.id as string,
+                OperationType.DELETE
+            );
+            notes.value.splice(noteIndex, 1);
+        }
+    }
+
+    function setNotesError(resultError: ResultError) {
+        error.value = resultError;
+        statusMessage.value = resultError.resultMessage;
+        status.value = LoadStatus.ERROR;
+    }
+
+    // Helpers
+    function handleError(
+        hdid: string,
+        resultError: ResultError,
+        errorType: ErrorType
+    ): void {
+        const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+
+        logger.error(`ERROR: ${JSON.stringify(resultError)}`);
+        setNotesError(resultError);
+
+        if (resultError.statusCode === 429) {
+            if (errorType === ErrorType.Retrieve) {
+                errorStore.setTooManyRequestsWarning("page");
+            } else if (errorType === ErrorType.Delete) {
+                errorStore.setTooManyRequestsError("page");
+            } else {
+                errorStore.setTooManyRequestsError("noteEditModal");
+            }
+        } else {
+            errorStore.addError(
+                errorType,
+                ErrorSourceType.Note,
+                resultError.traceId
+            );
+        }
+    }
+
+    // Actions
+    function retrieveNotes(params: { hdid: string }): Promise<void> {
+        const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+
+        const userNotes: UserNote[] = notes.value;
+        if (status.value === LoadStatus.LOADED) {
+            logger.debug(`Notes found stored, not querying!`);
+            return Promise.resolve();
+        } else {
+            logger.debug(`Retrieving User notes`);
+            setNotesRequested();
+            return noteService
+                .getNotes(params.hdid)
+                .then((result) => {
+                    if (result.resultStatus === ResultType.Success) {
+                        setRetrievedNotes(result.resourcePayload);
+                    } else {
+                        logger.error(
+                            `Notes retrieval failed! ${JSON.stringify(result)}`
+                        );
+                        if (result.resultError) {
+                            throw result.resultError;
+                        }
+                    }
+                })
+                .catch((error: ResultError) => {
+                    handleError(params.hdid, error, ErrorType.Retrieve);
+                    throw error;
+                });
+        }
+    }
+
+    function createNote(params: {
+        hdid: string;
+        note: UserNote;
+    }): Promise<UserNote | undefined> {
```

Comment:
The signature should be simplified to `createNote(hdid: string, note: UserNote)`.